amqVivaceShortcuts.user.js

```diff
@@ -296,12 +306,25 @@ const optimizedShortcuts = (targets) => {
   let minPos = Infinity;
   let bestSubstring = "";
   let shortcuts = [];
+  let altShortcuts = [];
   let currentLength = 0;
 
-  for (const substring of sortedSubstrings) {
+  for (let substring of sortedSubstrings) {
     const newLength = substring.length;
 
     // Search for longer substrings only if there are not enough shortcuts yet, but display *all* the shortest ones
+    if (newLength > currentLength){
```

Comment:
The condition should be checked also before expanding the `shortcuts` array.
```suggestion
    // Search for longer substrings only if there are not enough shortcuts yet, but display *all* the shortest ones
    if (newLength > currentLength && shortcuts.length >= NUM_SHORTCUTS) break;
```

Reply:
I believe that the reason that I left out this shortcut check here was that I wanted to ensure that any shortcuts which contain a down arrow of the max length to be added (e.g. if the min shortcuts is 2 and "abc" is a shortcut, but "bcde" and "bcd↓" are shortcuts. This loop would is used to add "bcd↓" after bcde, even if we've reached the min shortcut count of 2 since it is still a 4 character shortcut.) That said, I could probably move that check to the end of the if statement rather than in a separate line below it.

Reply:
That makes sense. I guess you could just add a comment with that example, just as a reminder, and leave the check for exiting the loop in its original position.